e2e/pollers.go

```diff
@@ -73,7 +73,19 @@ func waitUntilPodReady(ctx context.Context, kube *Kubeclient, podName string, t
 			remaining := time.Until(deadline)
 			if currentLogTime.Sub(lastLogTime) > logInterval {
 				// this logs every 5 minutes to reduce spam, iterations of poller are continuning as normal.
-				t.Logf("pod %s status: %s time before timeout: %v", podName, pod.Status.Phase, remaining)
+				t.Logf("-- pod metadata --\n")
+				t.Logf("   Name: %s\n              Namespace: %s\n              Node: %s\n              Status: %s\n              Start Time: %s\n", pod.Name, pod.Namespace, pod.Spec.NodeName, pod.Status.Phase, pod.Status.StartTime)
```

Comment:
this data is nice, though are we also streaming pod container logs at any point? if not then might also be worth including in a log file alongside kubelet/cluster provisioning logs

Reply:
yeah I think streaming pod logs on failures would be a helpful next step

Reply:
From what I could tell we are not streaming and pod logs. I tried to get something to work but it kept sig faulting lol. Might be worth creating a backlog item for generating a file based on streaming pod container logs?

Reply:
Logs are likely to be empty for majority of such failures.

Reply:
FYI. Snippet from another place I used to extract debug info:

```
result.Pods = append(result.Pods, podDebugInfo{
			Name:                  pod.Name,
			Status:                pod.Status,
			ContainerLogs:         t.podLogs(ctx, pod, maxContainers, false),
			PreviousContainerLogs: t.podLogs(ctx, pod, maxContainers, true),
		})
	}
	return result
}

func (t *AgentTunnelKube) podLogs(ctx context.Context, pod v1.Pod, maxContainers int, previous bool) map[string]string {
	containers := make(map[string]string)
	for j, container := range pod.Spec.Containers {
		if j > maxContainers {
			break
		}
		containerLog, err := t.clientset.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, &v1.PodLogOptions{
			// without specifying TailLines, log head instead of tail is returned
			// important information is nearly always in the tail
			TailLines:  pointer.Int64(15),
			LimitBytes: pointer.Int64(40000), // safety limit, don't trust user input
			Container:  container.Name,
			Previous:   previous,
		}).DoRaw(ctx)
		if err != nil { //+gocover:ignore:block can't test it
			logger.Error().Str("namespace", t.AgentNamespace).Str("pod_name", pod.Name).
				Str("container_name", container.Name).Err(err).Msg("failed to get container logs")
			continue
		}
		containers[container.Name] = string(containerLog)
	}
	return containers
}
```